.github/workflows/patch.yaml

```diff
@@ -0,0 +1,132 @@
+name: Patch Images
+
+on:
+  schedule:
+    - cron: '0 2 */15 * *'  # Every 15 days at 2 a.m.
+#   push:
+#     branches:
+#       - main
+
+jobs:
+  setup:
+    runs-on: ubuntu-latest
+    outputs:
+        mymatrix: ${{ steps.dataStep.outputs.myoutput }}
+    steps:
+      - name: Checkout Repo
+        uses: actions/checkout@v4
+
+      # Retrieve image list via python (you can choose to include the last 3 tags by passing the '--include-last-3-tags' argument to the script)
+      - name: Export Image List With Python
+        id: dataStep
+        run: |
+          cd utilities
+          TARGETS=$(python3 image_list_json.py)
+          echo $TARGETS
+          echo "myoutput=$(jq -cn --argjson environments "$TARGETS" '{target: $environments}')" >> $GITHUB_OUTPUT
+
+  patch:
+    needs: setup
+    runs-on: ubuntu-latest
+    strategy:
+        fail-fast: false
+        matrix: ${{fromJson(needs.setup.outputs.mymatrix)}}
+    steps:
+    - name: Print image name
+      run: echo ${{matrix.target}}
+
+    - name: Install Cosign
+      uses: sigstore/cosign-installer@v3.1.1
+      with:
+        cosign-release: 'v2.2.3'
+
+    - name: Generate Trivy Report
+      uses: aquasecurity/trivy-action@69cbbc0cbbf6a2b0bab8dcf0e9f2d7ead08e87e4
+      with:
+        scan-type: 'image'
+        format: 'json'
+        output: 'report.json'
+        ignore-unfixed: true
+        vuln-type: 'os'
+        severity: 'MEDIUM,HIGH,CRITICAL'
+        image-ref: ${{ matrix.target }}
```

Comment:
Harbor already scans periodically (daily IIRC) all the images that are stored there.

Have you evaluated using the vulns information available in Harbor instead of re-scanning all these images again?

This may be useful on how to get vuln information from Harbor with Python:
https://github.com/ralgozino/harborster

Reply:
Copacetic requires the JSON report of the Trivy scan to function properly. 
Additionally, I believe it would be preferable to scan the images directly within the pipeline runtime to obtain an accurate snapshot of vulnerabilities at the exact moment of scanning.
This is because we can then include reports in *OpenVex* format, potentially using them to attest the images in the future.
Therefore, they must be generated by the same pipeline conducting the scans (pipeline artifacts/metadata).

Reply:
> Additionally, I believe it would be preferable to scan the images directly within the pipeline runtime to obtain an accurate snapshot of vulnerabilities at the exact moment of scanning.

I don't fully agree on this, I'd rather like to see the same vulns here than on the registry. Besides that scanning all the images again is a waste of compute.

Anyway, if the tool you are proposing requires the output from trivy then there's no much we can do.